crypto/rand/drbg_ctr.c

```diff
@@ -199,20 +200,23 @@ __owur static int ctr_df(RAND_DRBG_CTR *ctr,
         || !ctr_BCC_final(ctr))
         return 0;
     /* Set up key K */
-    if (!EVP_CipherInit_ex(ctr->ctx, ctr->cipher, NULL, ctr->KX, NULL, 1))
+    if (!EVP_CipherInit_ex(ctr->ctx_ecb,
+                           ctr->cipher_ecb, NULL, ctr->KX, NULL, 1)
+        || !EVP_CipherInit_ex(ctr->ctx_ctr,
+                              ctr->cipher_ctr, NULL, ctr->KX, NULL, 1))
```

Comment:
I feel here also initializing `ctx->ctr` is not needed. Only initializing at the end of `ctr_update` will be sufficient I feel.

Reply:
You are right in both cases. I pushed a fixup and updated the perf data in my previous post. It looks slightly better now.